lua/entities/gmod_wire_dhdd.lua

```diff
@@ -14,11 +14,33 @@ function ENT:Initialize()
 	self.Outputs = WireLib.CreateOutputs( self, { "Memory [ARRAY]", "Size" } )
 	self.Inputs = WireLib.CreateInputs( self, { "Data [ARRAY]", "Clear", "AllowWrite" } )
 
-	self.Memory = {}
+	self:SetMemory()
 	self.ROM = false
 	self.AllowWrite = true
+end
+
+function ENT:SetMemory(Memory)
```

Comment:
I did it because on the old code the value of self.Memory was replaced too, and if it gets replaced then the new one wouldn't have a proxy.